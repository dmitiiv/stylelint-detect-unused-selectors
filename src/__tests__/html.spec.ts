import path from 'path';
import stylelint from 'stylelint';
import { parseResult } from './__helpers__/parse-result';

const fixturesRoot = path.join(__dirname, '..', '..', 'examples', 'html');
const configFilePath = path.join(fixturesRoot, '.stylelintrc.js');

test('Disabllow selectors that are not used in a HTML file', async (): Promise<
  void
> => {
  const options = {
    configFile: configFilePath,
    files: path.join(fixturesRoot, '*.css'),
  };

  const result = await stylelint.lint(options);

  expect(result.errored).toBe(true);
  expect(parseResult(result)).toMatchSnapshot();
});
