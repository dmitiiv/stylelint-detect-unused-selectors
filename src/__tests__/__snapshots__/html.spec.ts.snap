// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disallow selectors with error that are not used in a HTML file 1`] = `
{
  "deprecations": [],
  "errored": true,
  "invalidOptionWarnings": [],
  "parseErrors": [],
  "warnings": [
    {
      "column": 1,
      "endColumn": 2,
      "endLine": 29,
      "line": 27,
      "rule": "plugin/detect-unused-selectors",
      "severity": "error",
      "text": ".not-found is defined but doesn't match any elements in test.html. (plugin/detect-unused-selectors)",
    },
    {
      "column": 1,
      "endColumn": 2,
      "endLine": 34,
      "line": 32,
      "rule": "plugin/detect-unused-selectors",
      "severity": "error",
      "text": ".list-item:last-child + li is defined but doesn't match any elements in test.html. (plugin/detect-unused-selectors)",
    },
  ],
}
`;

exports[`Disallow selectors with warning that are not used in a HTML file 1`] = `
{
  "deprecations": [],
  "errored": false,
  "invalidOptionWarnings": [],
  "parseErrors": [],
  "warnings": [
    {
      "column": 1,
      "endColumn": 2,
      "endLine": 29,
      "line": 27,
      "rule": "plugin/detect-unused-selectors",
      "severity": "warning",
      "text": ".not-found is defined but doesn't match any elements in test.html. (plugin/detect-unused-selectors)",
    },
    {
      "column": 1,
      "endColumn": 2,
      "endLine": 34,
      "line": 32,
      "rule": "plugin/detect-unused-selectors",
      "severity": "warning",
      "text": ".list-item:last-child + li is defined but doesn't match any elements in test.html. (plugin/detect-unused-selectors)",
    },
  ],
}
`;
